app/controllers/api/plans_base_controller.rb

```diff
@@ -112,6 +112,18 @@ def plans_index_path
 
   protected
 
+  def new_masterize_plan(issuer, assoc)
+    if params[:id].blank? || issuer.send(assoc) == @plan
+      issuer.send("#{assoc}=", nil)
+    elsif @plan.nil?
+      raise ActiveRecord::RecordNotFound, "Plan doesn't exist"
+    else
+      issuer.send("#{assoc}=", @plan)
+    end
```

Comment:
I see some DRY opportunity here when comparing to `#generic_masterize_plan`

Reply:
`generic_masterize_plan` will be removed in a following PR (right after this one is merged), after all of the "old" selects have been replaced with this one. At which point there will be only one `#masterize_plan` method, that's why I don't think refactoring is worth it BUT any suggestions welcome.